snake-form-authentication-route-fix

kslo88 opened the pull request:
Resolves #4 #3

Sushanthbc commented on ios/Podfile.lock:
```diff
@@ -1,53 +1,82 @@
-  - Firebase/Core (4.13.0):
-    - FirebaseAnalytics (= 4.2.0)
-    - FirebaseCore (= 4.0.20)
+  - Firebase/Auth (5.6.0):
```
We upgraded firebase and google sign. 
Check step: I should pull your branch in my system and see if we something is breaking.

Sushanthbc commented on ios/Runner.xcodeproj/project.pbxproj:
```diff
@@ -21,6 +21,7 @@
 		97C146FE1CF9000F007C117D /* Assets.xcassets in Resources */ = {isa = PBXBuildFile; fileRef = 97C146FD1CF9000F007C117D /* Assets.xcassets */; };
```
Should we have this in .gitignore?

Sushanthbc commented on ios/Runner/GoogleService-Info.plist:
```diff
@@ -7,21 +7,21 @@
 	<key>AD_UNIT_ID_FOR_INTERSTITIAL_TEST</key>
```
This should be going to .gitignore 100%

Sushanthbc commented on lib/src/pages/form_snake.dart:
```diff
@@ -0,0 +1,604 @@
+part of dozer;
+
+
+class FormSnake extends StatelessWidget {
```
We can remove this entire class, this functionality already set at dozer_app.

Sushanthbc commented on lib/src/pages/form_snake.dart:
```diff
@@ -0,0 +1,604 @@
+
+
+// Create a Form Widget
+class MyCustomForm extends StatefulWidget {
```
We will change this method to SnakeForm

Sushanthbc commented on lib/src/pages/form_snake.dart:
```diff
@@ -0,0 +1,604 @@
+// Create a Form Widget
+class MyCustomForm extends StatefulWidget {
+  @override
+  MyCustomFormState createState() {
```
_SnakeFormState

kslo88 commented:
After taking a pull in "Form-with-required-fields" branch, build is failing
with below errors.

Failed to build iOS app
Error output from Xcode build:
↳
    ** BUILD FAILED **

Xcode's output:
↳
    === BUILD TARGET google_sign_in OF PROJECT Pods WITH CONFIGURATION
Debug ===
    In file included from
/Users/lohith/app_dev/Tools/flutter/.pub-cache/hosted/
pub.dartlang.org/firebase_auth-0.5.4/ios/Classes/FirebaseAuthPlugin.m:5:
    In file included from
/Users/lohith/app_dev/Tools/flutter/.pub-cache/hosted/
pub.dartlang.org/firebase_auth-0.5.4/ios/Classes/FirebaseAuthPlugin.h:5:
    In file included from
/Users/lohith/dozer_app/ios/Pods/Headers/Public/Flutter/Flutter/Flutter.h:37:
    In file included from
/Users/lohith/dozer_app/ios/Pods/Headers/Public/Flutter/Flutter/FlutterAppDelegate.h:11:

/Users/lohith/dozer_app/ios/Pods/Headers/Public/Flutter/Flutter/FlutterPlugin.h:80:42:
warning: 'UIUserNotificationSettings' is partial: introduced in iOS 8.0 -
Use UserNotifications Framework's UNNotificationSettings
[-Wunguarded-availability]

didRegisterUserNotificationSettings:(UIUserNotificationSettings*)notificationSettings;
                                             ^
    In module 'UIKit' imported from /Users/lohith/dozer_app/ios/Pods/Target
Support Files/firebase_auth/firebase_auth-prefix.pch:2:

/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator11.1.sdk/System/Library/Frameworks/UIKit.framework/Headers/UIUserNotificationSettings.h:43:12:
note: 'UIUserNotificationSettings' has been explicitly marked partial here
    @interface UIUserNotificationSettings : NSObject
               ^
    In file included from
/Users/lohith/app_dev/Tools/flutter/.pub-cache/hosted/
pub.dartlang.org/firebase_auth-0.5.4/ios/Classes/FirebaseAuthPlugin.m:5:
    In file included from
/Users/lohith/app_dev/Tools/flutter/.pub-cache/hosted/
pub.dartlang.org/firebase_auth-0.5.4/ios/Classes/FirebaseAuthPlugin.h:5:
    In file included from
/Users/lohith/dozer_app/ios/Pods/Headers/Public/Flutter/Flutter/Flutter.h:37:
    In file included from
/Users/lohith/dozer_app/ios/Pods/Headers/Public/Flutter/Flutter/FlutterAppDelegate.h:11:

/Users/lohith/dozer_app/ios/Pods/Headers/Public/Flutter/Flutter/FlutterPlugin.h:79:1:
note: annotate 'application:didRegisterUserNotificationSettings:' with an
availability attribute to silence
    - (void)application:(UIApplication*)application
    ^

/Users/lohith/dozer_app/ios/Pods/Headers/Public/Flutter/Flutter/FlutterPlugin.h:95:44:
warning: 'UIBackgroundFetchResult' is partial: introduced in iOS 7.0
[-Wunguarded-availability]
              fetchCompletionHandler:(void (^)(UIBackgroundFetchResult
result))completionHandler;
                                               ^
    In module 'UIKit' imported from /Users/lohith/dozer_app/ios/Pods/Target
Support Files/firebase_auth/firebase_auth-prefix.pch:2:

/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator11.1.sdk/System/Library/Frameworks/UIKit.framework/Headers/UIApplication.h:76:29:
note: 'UIBackgroundFetchResult' has been explicitly marked partial here
    typedef NS_ENUM(NSUInteger, UIBackgroundFetchResult) {
                                ^
    In file included from
/Users/lohith/app_dev/Tools/flutter/.pub-cache/hosted/
pub.dartlang.org/firebase_auth-0.5.4/ios/Classes/FirebaseAuthPlugin.m:5:
    In file included from
/Users/lohith/app_dev/Tools/flutter/.pub-cache/hosted/
pub.dartlang.org/firebase_auth-0.5.4/ios/Classes/FirebaseAuthPlugin.h:5:
    In file included from
/Users/lohith/dozer_app/ios/Pods/Headers/Public/Flutter/Flutter/Flutter.h:37:
    In file included from
/Users/lohith/dozer_app/ios/Pods/Headers/Public/Flutter/Flutter/FlutterAppDelegate.h:11:

/Users/lohith/dozer_app/ios/Pods/Headers/Public/Flutter/Flutter/FlutterPlugin.h:93:1:
note: annotate
'application:didReceiveRemoteNotification:fetchCompletionHandler:' with an
availability attribute to silence
    - (BOOL)application:(UIApplication*)application
    ^

/Users/lohith/dozer_app/ios/Pods/Headers/Public/Flutter/Flutter/FlutterPlugin.h:140:58:
warning: this block declaration is not a prototype [-Wstrict-prototypes]
                          completionHandler:(nonnull void
(^)())completionHandler;
                                                             ^
                                                              void

/Users/lohith/dozer_app/ios/Pods/Headers/Public/Flutter/Flutter/FlutterPlugin.h:148:49:
warning: 'UIBackgroundFetchResult' is partial: introduced in iOS 7.0
[-Wunguarded-availability]
        performFetchWithCompletionHandler:(void (^)(UIBackgroundFetchResult
result))completionHandler;
                                                    ^
    In module 'UIKit' imported from /Users/lohith/dozer_app/ios/Pods/Target
Support Files/firebase_auth/firebase_auth-prefix.pch:2:

/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator11.1.sdk/System/Library/Frameworks/UIKit.framework/Headers/UIApplication.h:76:29:
note: 'UIBackgroundFetchResult' has been explicitly marked partial here
    typedef NS_ENUM(NSUInteger, UIBackgroundFetchResult) {
                                ^
    In file included from
/Users/lohith/app_dev/Tools/flutter/.pub-cache/hosted/
pub.dartlang.org/firebase_auth-0.5.4/ios/Classes/FirebaseAuthPlugin.m:5:
    In file included from
/Users/lohith/app_dev/Tools/flutter/.pub-cache/hosted/
pub.dartlang.org/firebase_auth-0.5.4/ios/Classes/FirebaseAuthPlugin.h:5:
    In file included from
/Users/lohith/dozer_app/ios/Pods/Headers/Public/Flutter/Flutter/Flutter.h:37:
    In file included from
/Users/lohith/dozer_app/ios/Pods/Headers/Public/Flutter/Flutter/FlutterAppDelegate.h:11:

/Users/lohith/dozer_app/ios/Pods/Headers/Public/Flutter/Flutter/FlutterPlugin.h:147:1:
note: annotate 'application:performFetchWithCompletionHandler:' with an
availability attribute to silence
    - (BOOL)application:(UIApplication*)application
    ^
    In file included from
/Users/lohith/app_dev/Tools/flutter/.pub-cache/hosted/
pub.dartlang.org/firebase_auth-0.5.4/ios/Classes/FirebaseAuthPlugin.m:5:
    /Users/lohith/app_dev/Tools/flutter/.pub-cache/hosted/
pub.dartlang.org/firebase_auth-0.5.4/ios/Classes/FirebaseAuthPlugin.h:7:9:
fatal error: 'Firebase/Firebase.h' file not found
    #import "Firebase/Firebase.h"
            ^~~~~~~~~~~~~~~~~~~~~
    4 warnings and 1 error generated.
Could not build the application for the simulator.
Error launching application on iPhone X.

On Mon, Aug 27, 2018 at 11:00 AM, Sushanth BC <notifications@github.com>
wrote:

> *@Sushanthbc* commented on this pull request.
> ------------------------------
>
> In ios/Podfile.lock
> <https://github.com/Sushanthbc/dozer_app/pull/6#discussion_r212867315>:
>
> > @@ -1,53 +1,82 @@
>  PODS:
> -  - Firebase/Auth (4.13.0):
> -    - Firebase/Core
> -    - FirebaseAuth (= 4.6.1)
> -  - Firebase/Core (4.13.0):
> -    - FirebaseAnalytics (= 4.2.0)
> -    - FirebaseCore (= 4.0.20)
> +  - Firebase/Auth (5.6.0):
>
> We upgraded firebase and google sign.
> Check step: I should pull your branch in my system and see if we something
> is breaking.
> ------------------------------
>
> In ios/Runner.xcodeproj/project.pbxproj
> <https://github.com/Sushanthbc/dozer_app/pull/6#discussion_r212867391>:
>
> > @@ -21,6 +21,7 @@
>  		97C146FE1CF9000F007C117D /* Assets.xcassets in Resources */ = {isa = PBXBuildFile; fileRef = 97C146FD1CF9000F007C117D /* Assets.xcassets */; };
>
> Should we have this in .gitignore?
> ------------------------------
>
> In ios/Runner/GoogleService-Info.plist
> <https://github.com/Sushanthbc/dozer_app/pull/6#discussion_r212867478>:
>
> > @@ -7,21 +7,21 @@
>  	<key>AD_UNIT_ID_FOR_INTERSTITIAL_TEST</key>
>
> This should be going to .gitignore 100%
> ------------------------------
>
> In lib/src/pages/form_snake.dart
> <https://github.com/Sushanthbc/dozer_app/pull/6#discussion_r212867643>:
>
> > @@ -0,0 +1,604 @@
> +part of dozer;
> +
> +
> +class FormSnake extends StatelessWidget {
>
> We can remove this entire class, this functionality already set at
> dozer_app.
> ------------------------------
>
> In lib/src/pages/form_snake.dart
> <https://github.com/Sushanthbc/dozer_app/pull/6#discussion_r212868037>:
>
> > +    return MaterialApp(
> +      title: appTitle,
> +      home: Scaffold(
> +        appBar: AppBar(
> +            title: Text(appTitle)
> +        ),
> +        body: MyCustomForm(),
> +      ),
> +    );
> +
> +  }
> +}
> +
> +
> +// Create a Form Widget
> +class MyCustomForm extends StatefulWidget {
>
> We will change this method to SnakeForm
> ------------------------------
>
> In lib/src/pages/form_snake.dart
> <https://github.com/Sushanthbc/dozer_app/pull/6#discussion_r212868076>:
>
> > +      home: Scaffold(
> +        appBar: AppBar(
> +            title: Text(appTitle)
> +        ),
> +        body: MyCustomForm(),
> +      ),
> +    );
> +
> +  }
> +}
> +
> +
> +// Create a Form Widget
> +class MyCustomForm extends StatefulWidget {
> +  @override
> +  MyCustomFormState createState() {
>
> _SnakeFormState
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Sushanthbc/dozer_app/pull/6#pullrequestreview-149582864>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ARmIVvwko76AewvMz-WxnGUfBaYbH3omks5uU4P0gaJpZM4WMyTw>
> .
>



-- 
Regards,
Lohith

Sushanthbc commented:
Goto IOS folder you will see pod.lock delete it and pubspec.lock delete it

Finally do package get.

On Tue, 28 Aug 2018 at 10:46 AM, kslo88 <notifications@github.com> wrote:

> After taking a pull in "Form-with-required-fields" branch, build is failing
> with below errors.
>
> Failed to build iOS app
> Error output from Xcode build:
> ↳
> ** BUILD FAILED **
>
> Xcode's output:
> ↳
> === BUILD TARGET google_sign_in OF PROJECT Pods WITH CONFIGURATION
> Debug ===
> In file included from
> /Users/lohith/app_dev/Tools/flutter/.pub-cache/hosted/
> pub.dartlang.org/firebase_auth-0.5.4/ios/Classes/FirebaseAuthPlugin.m:5:
> In file included from
> /Users/lohith/app_dev/Tools/flutter/.pub-cache/hosted/
> pub.dartlang.org/firebase_auth-0.5.4/ios/Classes/FirebaseAuthPlugin.h:5:
> In file included from
>
> /Users/lohith/dozer_app/ios/Pods/Headers/Public/Flutter/Flutter/Flutter.h:37:
> In file included from
>
> /Users/lohith/dozer_app/ios/Pods/Headers/Public/Flutter/Flutter/FlutterAppDelegate.h:11:
>
>
> /Users/lohith/dozer_app/ios/Pods/Headers/Public/Flutter/Flutter/FlutterPlugin.h:80:42:
> warning: 'UIUserNotificationSettings' is partial: introduced in iOS 8.0 -
> Use UserNotifications Framework's UNNotificationSettings
> [-Wunguarded-availability]
>
>
> didRegisterUserNotificationSettings:(UIUserNotificationSettings*)notificationSettings;
> ^
> In module 'UIKit' imported from /Users/lohith/dozer_app/ios/Pods/Target
> Support Files/firebase_auth/firebase_auth-prefix.pch:2:
>
>
> /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator11.1.sdk/System/Library/Frameworks/UIKit.framework/Headers/UIUserNotificationSettings.h:43:12:
> note: 'UIUserNotificationSettings' has been explicitly marked partial here
> @interface UIUserNotificationSettings : NSObject
> ^
> In file included from
> /Users/lohith/app_dev/Tools/flutter/.pub-cache/hosted/
> pub.dartlang.org/firebase_auth-0.5.4/ios/Classes/FirebaseAuthPlugin.m:5:
> In file included from
> /Users/lohith/app_dev/Tools/flutter/.pub-cache/hosted/
> pub.dartlang.org/firebase_auth-0.5.4/ios/Classes/FirebaseAuthPlugin.h:5:
> In file included from
>
> /Users/lohith/dozer_app/ios/Pods/Headers/Public/Flutter/Flutter/Flutter.h:37:
> In file included from
>
> /Users/lohith/dozer_app/ios/Pods/Headers/Public/Flutter/Flutter/FlutterAppDelegate.h:11:
>
>
> /Users/lohith/dozer_app/ios/Pods/Headers/Public/Flutter/Flutter/FlutterPlugin.h:79:1:
> note: annotate 'application:didRegisterUserNotificationSettings:' with an
> availability attribute to silence
> - (void)application:(UIApplication*)application
> ^
>
>
> /Users/lohith/dozer_app/ios/Pods/Headers/Public/Flutter/Flutter/FlutterPlugin.h:95:44:
> warning: 'UIBackgroundFetchResult' is partial: introduced in iOS 7.0
> [-Wunguarded-availability]
> fetchCompletionHandler:(void (^)(UIBackgroundFetchResult
> result))completionHandler;
> ^
> In module 'UIKit' imported from /Users/lohith/dozer_app/ios/Pods/Target
> Support Files/firebase_auth/firebase_auth-prefix.pch:2:
>
>
> /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator11.1.sdk/System/Library/Frameworks/UIKit.framework/Headers/UIApplication.h:76:29:
> note: 'UIBackgroundFetchResult' has been explicitly marked partial here
> typedef NS_ENUM(NSUInteger, UIBackgroundFetchResult) {
> ^
> In file included from
> /Users/lohith/app_dev/Tools/flutter/.pub-cache/hosted/
> pub.dartlang.org/firebase_auth-0.5.4/ios/Classes/FirebaseAuthPlugin.m:5:
> In file included from
> /Users/lohith/app_dev/Tools/flutter/.pub-cache/hosted/
> pub.dartlang.org/firebase_auth-0.5.4/ios/Classes/FirebaseAuthPlugin.h:5:
> In file included from
>
> /Users/lohith/dozer_app/ios/Pods/Headers/Public/Flutter/Flutter/Flutter.h:37:
> In file included from
>
> /Users/lohith/dozer_app/ios/Pods/Headers/Public/Flutter/Flutter/FlutterAppDelegate.h:11:
>
>
> /Users/lohith/dozer_app/ios/Pods/Headers/Public/Flutter/Flutter/FlutterPlugin.h:93:1:
> note: annotate
> 'application:didReceiveRemoteNotification:fetchCompletionHandler:' with an
> availability attribute to silence
> - (BOOL)application:(UIApplication*)application
> ^
>
>
> /Users/lohith/dozer_app/ios/Pods/Headers/Public/Flutter/Flutter/FlutterPlugin.h:140:58:
> warning: this block declaration is not a prototype [-Wstrict-prototypes]
> completionHandler:(nonnull void
> (^)())completionHandler;
> ^
> void
>
>
> /Users/lohith/dozer_app/ios/Pods/Headers/Public/Flutter/Flutter/FlutterPlugin.h:148:49:
> warning: 'UIBackgroundFetchResult' is partial: introduced in iOS 7.0
> [-Wunguarded-availability]
> performFetchWithCompletionHandler:(void (^)(UIBackgroundFetchResult
> result))completionHandler;
> ^
> In module 'UIKit' imported from /Users/lohith/dozer_app/ios/Pods/Target
> Support Files/firebase_auth/firebase_auth-prefix.pch:2:
>
>
> /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator11.1.sdk/System/Library/Frameworks/UIKit.framework/Headers/UIApplication.h:76:29:
> note: 'UIBackgroundFetchResult' has been explicitly marked partial here
> typedef NS_ENUM(NSUInteger, UIBackgroundFetchResult) {
> ^
> In file included from
> /Users/lohith/app_dev/Tools/flutter/.pub-cache/hosted/
> pub.dartlang.org/firebase_auth-0.5.4/ios/Classes/FirebaseAuthPlugin.m:5:
> In file included from
> /Users/lohith/app_dev/Tools/flutter/.pub-cache/hosted/
> pub.dartlang.org/firebase_auth-0.5.4/ios/Classes/FirebaseAuthPlugin.h:5:
> In file included from
>
> /Users/lohith/dozer_app/ios/Pods/Headers/Public/Flutter/Flutter/Flutter.h:37:
> In file included from
>
> /Users/lohith/dozer_app/ios/Pods/Headers/Public/Flutter/Flutter/FlutterAppDelegate.h:11:
>
>
> /Users/lohith/dozer_app/ios/Pods/Headers/Public/Flutter/Flutter/FlutterPlugin.h:147:1:
> note: annotate 'application:performFetchWithCompletionHandler:' with an
> availability attribute to silence
> - (BOOL)application:(UIApplication*)application
> ^
> In file included from
> /Users/lohith/app_dev/Tools/flutter/.pub-cache/hosted/
> pub.dartlang.org/firebase_auth-0.5.4/ios/Classes/FirebaseAuthPlugin.m:5:
> /Users/lohith/app_dev/Tools/flutter/.pub-cache/hosted/
> pub.dartlang.org/firebase_auth-0.5.4/ios/Classes/FirebaseAuthPlugin.h:7:9:
> fatal error: 'Firebase/Firebase.h' file not found
> #import "Firebase/Firebase.h"
> ^~~~~~~~~~~~~~~~~~~~~
> 4 warnings and 1 error generated.
> Could not build the application for the simulator.
> Error launching application on iPhone X.
>
> On Mon, Aug 27, 2018 at 11:00 AM, Sushanth BC <notifications@github.com>
> wrote:
>
> > *@Sushanthbc* commented on this pull request.
> > ------------------------------
> >
> > In ios/Podfile.lock
> > <https://github.com/Sushanthbc/dozer_app/pull/6#discussion_r212867315>:
> >
> > > @@ -1,53 +1,82 @@
> > PODS:
> > - - Firebase/Auth (4.13.0):
> > - - Firebase/Core
> > - - FirebaseAuth (= 4.6.1)
> > - - Firebase/Core (4.13.0):
> > - - FirebaseAnalytics (= 4.2.0)
> > - - FirebaseCore (= 4.0.20)
> > + - Firebase/Auth (5.6.0):
> >
> > We upgraded firebase and google sign.
> > Check step: I should pull your branch in my system and see if we
> something
> > is breaking.
> > ------------------------------
> >
> > In ios/Runner.xcodeproj/project.pbxproj
> > <https://github.com/Sushanthbc/dozer_app/pull/6#discussion_r212867391>:
> >
> > > @@ -21,6 +21,7 @@
> > 97C146FE1CF9000F007C117D /* Assets.xcassets in Resources */ = {isa =
> PBXBuildFile; fileRef = 97C146FD1CF9000F007C117D /* Assets.xcassets */; };
> >
> > Should we have this in .gitignore?
> > ------------------------------
> >
> > In ios/Runner/GoogleService-Info.plist
> > <https://github.com/Sushanthbc/dozer_app/pull/6#discussion_r212867478>:
> >
> > > @@ -7,21 +7,21 @@
> > <key>AD_UNIT_ID_FOR_INTERSTITIAL_TEST</key>
> >
> > This should be going to .gitignore 100%
> > ------------------------------
> >
> > In lib/src/pages/form_snake.dart
> > <https://github.com/Sushanthbc/dozer_app/pull/6#discussion_r212867643>:
> >
> > > @@ -0,0 +1,604 @@
> > +part of dozer;
> > +
> > +
> > +class FormSnake extends StatelessWidget {
> >
> > We can remove this entire class, this functionality already set at
> > dozer_app.
> > ------------------------------
> >
> > In lib/src/pages/form_snake.dart
> > <https://github.com/Sushanthbc/dozer_app/pull/6#discussion_r212868037>:
> >
> > > + return MaterialApp(
> > + title: appTitle,
> > + home: Scaffold(
> > + appBar: AppBar(
> > + title: Text(appTitle)
> > + ),
> > + body: MyCustomForm(),
> > + ),
> > + );
> > +
> > + }
> > +}
> > +
> > +
> > +// Create a Form Widget
> > +class MyCustomForm extends StatefulWidget {
> >
> > We will change this method to SnakeForm
> > ------------------------------
> >
> > In lib/src/pages/form_snake.dart
> > <https://github.com/Sushanthbc/dozer_app/pull/6#discussion_r212868076>:
> >
> > > + home: Scaffold(
> > + appBar: AppBar(
> > + title: Text(appTitle)
> > + ),
> > + body: MyCustomForm(),
> > + ),
> > + );
> > +
> > + }
> > +}
> > +
> > +
> > +// Create a Form Widget
> > +class MyCustomForm extends StatefulWidget {
> > + @override
> > + MyCustomFormState createState() {
> >
> > _SnakeFormState
> >
> > —
> > You are receiving this because you authored the thread.
> > Reply to this email directly, view it on GitHub
> > <
> https://github.com/Sushanthbc/dozer_app/pull/6#pullrequestreview-149582864
> >,
> > or mute the thread
> > <
> https://github.com/notifications/unsubscribe-auth/ARmIVvwko76AewvMz-WxnGUfBaYbH3omks5uU4P0gaJpZM4WMyTw
> >
> > .
> >
>
>
>
> --
> Regards,
> Lohith
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Sushanthbc/dozer_app/pull/6#issuecomment-416454043>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ALjYpeARiOno3fWStWZSefVe8wirkWPUks5uVNI4gaJpZM4WMyTw>
> .
>

Sushanthbc commented on lib/src/utli/authentication.dart:
```diff
@@ -27,9 +27,11 @@ Future<FirebaseUser> signInWithGoogle() async {
 }
 
-Future<Null> signOutWithGoogle() async {
+Future<GoogleSignInAccount> signOutWithGoogle() async {
```
Should we return GoogleSignInAccount back?
I think simply doing Signout? should take care.

kslo88 replied:
I added sign out button for testing and it calls _auth.signOut(). But it wasn't working.

Check the authentication dart file. I added return disconnect(), but this works sometimes, doesn't work sometimes. Need to read more on this.

Sushanthbc commented on lib/src/pages/list_snakes.dart:
```diff
@@ -31,54 +39,41 @@ class SnakesListView extends StatefulWidget{
 class SnakesListViewState extends State<SnakesListView>{
 
-  Future<List<User>> _getSnakesList() async {
+  Future<List<SnakeInfo>> _getSnakesList() async {
```
This API call should handle promise, you are just testing for positive condition, what if the API returns any other status code like 400/500 etc.
I think the best way is to put it in promise. Use .then and .catchError.
https://www.dartlang.org/tutorials/language/futures

kslo88 replied:
This is being handled. I'm doing a check if the response has data or error. In case of error, currently I'm displaying a text with the response error message. Will change that to make it user understandable message.

Sushanthbc commented on lib/src/pages/list_snakes.dart:
```diff
@@ -31,54 +39,41 @@ class SnakesListView extends StatefulWidget{
-        snake_sex: data[i]["snake_sex"],
-        rescue_date_time: data[i]["rescue_date_time"]
+  List<SnakeInfo> _createSnakesList(List dataFromServer){
+    List<SnakeInfo> list = new List();
```
I think its much cleaner to use foreach instead of for
```
for (SnakeInfo tuple in dataFromServer) {
        snakeLength: tuple["snake_length"]
        ....
```

Sushanthbc commented:
Let's stop pushing commits to this PR.
We will just make the necessary changes as per code review.
Merge it to master.
